### PR TITLE
load cloudgov-style css, js, etc from node_modules

### PR DESCRIPTION
**Pull request by jseppi**
My attempt to use cloudgov-style from npm. You must run `npm install` to install the cloudgov-style module (any CI builds will need to be updated to do this). Hopefully this helps keep up-to-date with the latest cloudgov-style without having to build and manually copy files over from that project. 

Caveat: I'm not a rubyist so I do not know if what I've done here is anywhere close to correct or best practice.

Basically what I've done is
- Added `node_modules/cloudgov-style/css` and `node_modules/cloudgov-style/js` to Sinatra's assets pipeline. This allows inclusion of `cloudgov-style.css` and `cloudgov-style.js` in `app.css` and `app.js`.
- Added Sinatra routes to `/img/:file` and `/font/:file` that look for images and fonts in the respective `node_modules/cloudgov-style` folders.
- Changed `public/img/` to `public/images/` to avoid the path collision with the cloudgov-style paths in the previous bullet.

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -22,12 +22,17 @@ def initialize
...
+  cloudgov_path = File.join(root, 'node_modules', 'cloudgov-style')
   set :assets, Sprockets::Environment.new(root)
 
   configure do
     assets.append_path File.join(root, 'assets', 'stylesheets')
     assets.append_path File.join(root, 'assets', 'javascripts')
 
+    # cloudgov-style css and js. must be installed before running via `npm install`
+    assets.append_path File.join(cloudgov_path, 'css')
+    assets.append_path File.join(cloudgov_path, 'js')
```
Comment (juliasolorzano) [inline app.rb]: Not sure we need to add the `cg-style` js files, as they only include analytics at this point. Would be good to know if we want to track the same users for compliance viewer with this same GA #.

**Comment by jseppi** [comment]
Anyone else able to review? @afeld @DavidEBest

File: app.rb [hunk 2]
```diff
@@ -43,6 +47,14 @@ def initialize
     cache_control :public, :must_revalidate, :max_age => 60
   end
 
+  get '/img/:file' do |file|
+    send_file File.join(cloudgov_path, 'img', file)
+  end
+
+  get '/font/:file' do |file|
+    send_file File.join(cloudgov_path, 'font', file)
+  end
```
Comment (afeld) [inline app.rb]: Couldn't we do `assets.append_path` with these as well rather than having them as routes?

File: app.rb [hunk 3]
```diff
@@ -22,12 +22,16 @@ def initialize
...
   end
 
+  cloudgov_path = File.join(root, 'node_modules', 'cloudgov-style')
   set :assets, Sprockets::Environment.new(root)
 
   configure do
     assets.append_path File.join(root, 'assets', 'stylesheets')
     assets.append_path File.join(root, 'assets', 'javascripts')
 
+    # cloudgov-style css and js. must be installed before running via `npm install`
```
Comment (afeld) [inline app.rb]: Maybe we can add a line to check for the directory being present, and have it give this as an error message if it isn't?

**Comment by DavidEBest** [comment]
This looks pretty good to me. The routes for `img` and `font` give me pause, but I love the chance to remove all those files from the repo. 

The README should change to mention the need to run `npm install`.

**Comment by afeld** [comment]
Also, you'll need to add `npm install` to the Travis config.

https://docs.travis-ci.com/user/installing-dependencies

File: app.rb (hunk 2 above)
Comment (jseppi) [inline app.rb]: I'm not familiar with how Sinatra assets work. I tried adding those to assets (code below), which didn't work. It looks like the assets pipeline is for concatenating/minifying/etc, with the results being available through some special tags (like `<%= javascript_path 'app' %>` in `layout.erb for example).

``` ruby
assets.append_path File.join(cloudgov_path, 'img')
assets.append_path File.join(cloudgov_path, 'font')
```

File: app.rb (hunk 3 above)
Comment (jseppi) [inline app.rb]: Where do you think would be best to add this check, in the app code itself? Or perhaps in `config.ru`?

**Comment by jseppi** [comment]
Thanks for the pairing help, @afeld

**Pull request merged**